Ported from CSS to SASS using webpack + some documentation updates to adjust changes

code-n-chill opened the pull request:
<!-- Add a description of the pull request underneath this line. -->
Used Webpack 4 to move from CSS to SASS, as webpack folder is added which contains the config files as well as the SASS and JS files which would be compiled during lektor runtime.
The built CSS and JS files are configured to be stored in the asset folder.

<!-- If your pull request addresses an existing GitHub issue, replace the XX below with the issue number. -->

Closes #26  

<!-- Please follow this checklist and put an x in each of the boxes, like this: [x]. It will ensure that our team takes your pull request seriously. -->

**Checklist**  
- [x] My pull request has a descriptive title (not a vague title like `Update index.md`)
- [x] My pull request targets the `master` branch of creativecommons.github.io-source.
- [x] I tried building the site using Lektor locally and verified that there are no build errors.

<!-- Make sure you read and understand the following attestation. -->

Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.

hugosolar commented:
Would be ideal if you can fix this:
- There is a lack of use of Sass variables mostly for colors.
- There is a lot of elements that could be nested

code-n-chill commented:
@hugosolar, used variables for colors, nested elements and used mixins wherever possible. Please review and tell me if there is something else to fix.

kgodey commented on webpack/package.json:
```diff
@@ -0,0 +1,28 @@
+{
+  "name": "lektor-example",
```
could you rename this something less generic?

code-n-chill replied:
changed to: creative-commons

kgodey replied:
could you change to creative-commons-open-source since that's the title of the site?

code-n-chill replied:
[https://github.com/creativecommons/creativecommons.github.io-source/pull/31/commits/a605323a3487aa76c652ab46339d871772f43fe9](url)

code-n-chill replied:
done!

code-n-chill replied:
https://github.com/creativecommons/creativecommons.github.io-source/pull/31/commits/a605323a3487aa76c652ab46339d871772f43fe9

kgodey commented on webpack/sass/main.scss:
```diff
@@ -0,0 +1,165 @@
+
```
please remove extra newlines

code-n-chill replied:
done

kgodey commented on webpack/sass/main.scss:
```diff
@@ -0,0 +1,165 @@
+
+
+$orange1 : #EE5B32;
```
could we name these colors more descriptively, the current variables just describe the color and the variable will have to be renamed if the colors change. it's also difficult to understand what the difference between "orange1" and "orange2" is.

see https://css-tricks.com/what-do-you-name-color-variables/ or http://geoffgraham.me/naming-sass-color-variables/ (there are a bunch of them, these are just the first two I found) for some best practices around naming colors.

code-n-chill replied:
used:
brand-theme:
brand-theme-light:
brand-theme-dark:
etc.

kgodey commented:
@brenoferreira could you review the webpack portion of this PR?

hugosolar commented on webpack/sass/main.scss:
```diff
@@ -0,0 +1,170 @@
+  .navbar-toggler {
+    border-color: $white;
+  }
+  .navbar-nav .nav-item.active .nav-link {
```
This line should be nested, because up in the code you still have .nav-item elements

code-n-chill replied:
[https://github.com/creativecommons/creativecommons.github.io-source/pull/31/commits/a48e53bab9ce7f2531c6f3dce5d9ce5c61706f8b](url)

code-n-chill replied:
https://github.com/creativecommons/creativecommons.github.io-source/pull/31/commits/a48e53bab9ce7f2531c6f3dce5d9ce5c61706f8b

brenoferreira commented on webpack/js/main.js:
```diff
@@ -7,3 +7,4 @@ $(document).ready(function() {
         }
     });
 });
+//
```
Could you add a newline at the end of file?

code-n-chill replied:
Did it with the last commit. It's weird coz there is a newline in my local machine but when i am seeing it in github, there is no newline at the end. Can you please check?

code-n-chill replied:
It's weird that in the last commit I have added a new line but when I view the commit, it doesn't show a newline and when i again try to edit it within github, new line appears.

code-n-chill replied:
Can you please check?

brenoferreira commented on webpack/sass/main.scss:
```diff
@@ -0,0 +1,170 @@
+  .text-muted {
+    margin-left: 2rem;
+  }
+}
```
Missing newline at the end of file here too

brenoferreira commented on webpack/webpack.config.js:
```diff
@@ -0,0 +1,37 @@
+      }
+    ]
+  }
+};
```
And here

code-n-chill replied:
[https://github.com/creativecommons/creativecommons.github.io-source/pull/31/commits/a48e53bab9ce7f2531c6f3dce5d9ce5c61706f8b](url)

code-n-chill commented on webpack/sass/main.scss:
```diff
@@ -2,41 +2,50 @@ body {
+
 .navbar {
-  background-image:linear-gradient(90deg,#EE5B32,#FB7928,#EE5B32);
+  background-image: linear-gradient(90deg, #EE5B32, #FB7928, #EE5B32);
```
I think the second color (#FB7928) can't be obtained from the lighter function as the RGB ratios in both are different. Moreover, the navbar looks better with current config, here attaching both the screenshots where there is current navbar and other obtained by lightening the "main color" by 5%.

![Screenshot from 2019-03-12 23-48-37](https://user-images.githubusercontent.com/21370078/54225819-8a1ef500-4522-11e9-920c-c19ae6f14610.png)
![Screenshot from 2019-03-12 23-48-40](https://user-images.githubusercontent.com/21370078/54225820-8a1ef500-4522-11e9-9284-54b0cdec55c9.png)

code-n-chill replied:
As for now, I haven't changed it in the new commits. Just tell me what to do and I'll change it,

code-n-chill commented on webpack/sass/main.scss:
```diff
@@ -2,41 +2,50 @@ body {
   padding-bottom: 3rem;
 }
+
 .main-header .navbar-brand {
```
done!

code-n-chill commented on webpack/sass/main.scss:
```diff
@@ -75,49 +88,60 @@ h2 {
-  background: #e15803!important;
-}
-.navbar-dark .navbar-nav .nav-item .nav-link {
-  color: white;
```
done!

code-n-chill commented on webpack/sass/main.scss:
```diff
@@ -2,41 +2,50 @@ body {
 }
+
 .main-header {
   background-color: #EE5B32;
```
I already changed these in new commits.

code-n-chill commented on webpack/sass/main.scss:
```diff
@@ -0,0 +1,186 @@
+  border: none;
+}
+
+.nav-item, .dropdown-item {
```
https://github.com/creativecommons/creativecommons.github.io-source/pull/31/commits/1682ee9b23dc17b93b1336440939352e1d851e71

kgodey commented:
@code-n-chill this looks good, could you merge the latest master branch in and fix the merge conflicts? I'll merge it in once you do that.

code-n-chill commented:
The conflict is arising due to: https://github.com/creativecommons/creativecommons.github.io-source/pull/36
Through webpack, style.css is built at run time. I have deleted style.css in my pull request as it will be created by style.scss.
Since I have deleted style.css in the pull request and it was modified in the meantime on master, the conflict is arising. Reverting the above pull request can resolve the conflict.

kgodey commented:
@code-n-chill I had to revert because the styling was broken, see screenshot in https://github.com/creativecommons/creativecommons.github.io-source/pull/38. Please fix and open a new PR.

code-n-chill commented:
@kgodey, I forgot to build the package before the last commit. My bad :-( . Please check the new PR.
